Classes/Middleware/CookieDomainMiddleware.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the "headless" Extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.md file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace FriendsOfTYPO3\Headless\Middleware;
+
+use FriendsOfTYPO3\Headless\Utility\HeadlessFrontendUrlInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Log\LoggerInterface;
+use TYPO3\CMS\Core\Http\NormalizedParams;
+use TYPO3\CMS\Core\Site\SiteFinder;
+
+class CookieDomainMiddleware implements MiddlewareInterface
```

Comment:
Also can you write a unit test for it? if not please add @codeCoverageIgnore tag to the class for now.

Reply:
I can try.

We do not write tests on our projects. Which is why I lack experience here.

Reply:
Hi, it's ok, do not worry. Please add requested changes & add just flag @codeCoverageIgnore for now, so we can merge feature. I will add test for it later on.